wallet/src/main/java/com/android/identity_credential/wallet/presentation/OpenID4VPPresentationActivity.kt

```diff
@@ -894,22 +894,48 @@ private class TimeChecks : JWTClaimsSetVerifier<SecurityContext> {
     }
 }
 
-private fun getFormat(clientMetadata: JsonObject): CredentialFormat {
-    // Search for these strings, in order.
+private fun getFormat(
+    clientMetadata: JsonObject,
+    presentationDefinition: JsonObject
+): CredentialFormat {
     val acceptedFormats = listOf(
         "mso_mdoc" to CredentialFormat.MDOC_MSO,
         "jwt_vc" to CredentialFormat.SD_JWT_VC,
         "vc+sd-jwt" to CredentialFormat.SD_JWT_VC)
-    clientMetadata["vp_formats"]?.let {
-        val vpFormats = it.jsonObject.keys
-        for ((format, credentialFormat) in acceptedFormats) {
-            if (vpFormats.contains(format)) {
-                return credentialFormat
+
+    // if the clientMetadata has a exactly one format, return that
+    if (clientMetadata.containsKey("vp_formats")) {
+        val vpFormats = clientMetadata["vp_formats"]!!.jsonObject.keys
+        if (vpFormats.size == 1) {
+            for ((format, credentialFormat) in acceptedFormats) {
+                if (vpFormats.contains(format)) {
+                    return credentialFormat
+                }
             }
+            throw IllegalArgumentException("No supported formats found in: ${vpFormats.sorted().joinToString()}")
+        }
+    }
+
+    // else, use presentation_definition -> input_descriptors -> format
+    val inputDescriptors = presentationDefinition["input_descriptors"]!!.jsonArray
+    val vpFormats = HashSet<String>()
+
+    for (inputDescriptor: JsonElement in inputDescriptors) {
+        val inputDescriptorObj = inputDescriptor.jsonObject
+        val formatName = inputDescriptorObj["format"]?.jsonObject?.keys?.first() ?: "mso_mdoc"
+        if (!validFormats.contains(formatName)) {
```

Comment:
This feels fragile. Just because there's a format we don't recognize doesn't mean there are no other entries that we would recognize. Why throw an exception for this?

Reply:
makes sense, I threw an exception bc that's what was done below in `createPresentationSubmission` but adjusted `getFormat` and `createPresentationSubmission` so they simply ignore unexpected formats